### ISSUE TITLE
Create script to generate CHANGELOG.md (for demos & Whats Newsletter)

### ISSUE THREAD

**Issue by LZoog** [mozilla/experimenter#4809]
This issue might need discussion.

This will help with the Whats Newsletter as well as what we should demo.

We may want to run the script at sprint close - it probably just needs to list out all the titles of closed PRs from the last time the script was ran.

Related: #3669

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1071)
┆Issue Number: EXP-1071

**Comment by k88hudson** [comment]
Yeah this seems cool, is the idea it would be edited collectively? Like everyone sort of picks things to call out?

**Comment by data-sync-user** [comment]
➤ Jody Heavener commented:

could hook this up to a GitHub Action on a cron and send the suggestions over to a Slack channel

**Comment by LZoog** [comment]
Oh I like Jody's suggestion! Though I was thinking I'd like a PR for changes in a `CHANGELOG.md` file so we can have it for historical purposes, but maybe we don't necessarily need that?

@k88hudson Yeah I suppose after the changelog is posted somewhere we could make it a point to discuss (in Slack or even take 5 minutes in the retro meeting) on who is planning to demo what, as @lmorchard, @emtwo, and I were able to divvy up the work pretty easily for our shared demo, though since demos are on Thursday, I wonder if the script should be ran every other Wednesday morning or something.

**Comment by pdehaan** [comment]
Doesn't look like we use tags/versions in this repo anymore, so I don't think we can use something like [**conventional-changelog**](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli) here.

```sh
npx conventional-changelog-cli -p angular -i CHANGELOG.md -s -r 0
```

I was trying to remember what we used in FxA a lifetime ago, but it seems to either be wrapped in some shell script stuff now or as some part of `grunt version` which I can't quickly see the code for (so might be buried in some other task).

Still struggling to figure out if it'd be easier to do via the GitHub Pull Requests API, or figure out the Jira API for recently closed tickets (which one is cleaner and shows the specific ticket number format that you might want). I'm curious if there'd be a bunch of PRs in GitHub that don't have matching Jira tickets if they were just small maintenance things that might not show up in one CHANGELOG depending on how we generate that file.

**Comment by pdehaan** [comment]
I just realized my 🧁 leaderboard script is basically a PR changelog tool (with goofy table formatting). Here's a list of all the merged PRs in the past ~2 weeks:

```sh
> node changelog

#4808    3/12/2021      jaredlockhart   chore(nimbus): use curl/unzip instead of gitsubmodule to track jetstream-config
#4792    3/11/2021      jodyheavener    fixes #4684 feat(project): modify KintoClient to cache collection response on get_collection calls
#4788    3/10/2021      jaredlockhart   chore(nimbus): change jetstream-config to track main branch
#4787    3/10/2021      jaredlockhart   fix #4571 chore(nimbus): remove probesets
#4784    3/10/2021      jodyheavener    fixes #4752 feat(nimbus): filter the outcomes available for selection by experiment application
#4783    3/10/2021      emtwo           Fix #4669 - feat(visualization) Add weekly results prior to the end of an experiment.
#4779     3/9/2021      lmorchard       fix #4757 feat(nimbus): add outcomes to V6 API
#4760     3/8/2021      jodyheavener    feat(nimbus): tweak results sidebar styling
#4759    3/10/2021      LZoog           fixes #4572 feat(nimbus): Update probe set references to outcome
#4756     3/4/2021      jaredlockhart   fix #4570 feat(nimbus): add primary/secondary outcomes to nimbus expe…
#4754     3/3/2021      jaredlockhart   chore(project): allow nested .vscode folders
#4751     3/3/2021      jaredlockhart   fix #4750 chore(project): use makefile to build deploy container
#4749     3/3/2021      jodyheavener    chore(project): check in select vscode settings files
#4748     3/3/2021      lmorchard       fixes #4543 fix(nimbus): ensure dummy feature configs have schemas
#4746     3/2/2021      jaredlockhart   fix #4569 feat(nimbus): load outcomes from jetstream config
#4743     3/3/2021      emtwo           Fix #4507: feat(visualization) Add treatment and control definitions to the top of the page.
#4742     3/1/2021      jrbenny35       chore(project): Switch to release firefox for integration tests
#4737     3/1/2021      jaredlockhart   chore(project): disable intermittently failing integration tests
```

<details>
<summary>script</summary>

```js
const { Octokit } = require("@octokit/rest");

const octokit = new Octokit({
  auth: process.env.GITHUB_TOKEN,
});

main("mozilla", "experimenter", new Date("Mar 1, 2021"));

async function main(owner, repo, startDate, endDate) {
  startDate = startDate || new Date();
  endDate = endDate || new Date();
  const { data: pulls } = await octokit.pulls.list({
    owner,
    repo,
    state: "closed",
    per_page: 100,
  });
  const changelogPulls = pulls.filter((pr) => {
    const mergedAt = new Date(pr.merged_at);
    return (
      pr.user.type !== "Bot" &&
      mergedAt &&
      mergedAt >= startDate &&
      mergedAt < endDate
    );
  });
  // console.log(JSON.stringify(changelogPulls, null, 2));

  for (const pr of changelogPulls) {
    const mergedAt = new Date(pr.merged_at).toLocaleDateString();
    console.log([
      `#${pr.number}`,
      mergedAt.padStart(10, " "),
      pr.user.login.padEnd(12, " "),
      pr.title.trim()
    ].join("\t"));
  }
}
```
</details>
